template/src/pages/Mint.jsx

```diff
@@ -0,0 +1,105 @@
+import React, { useEffect, useState } from "react";
+import { abi as contractABI } from "../contracts/NFTCollection/abi.json";
+import { address } from "../contracts/NFTCollection/address.json";
+
+import { getSignedContract } from "../utils/common.js";
+
+import { ToastContainer, toast } from "react-toastify";
+import "react-toastify/dist/ReactToastify.css";
+
+export default function Mint() {
+  const [imageUrl, setImageUrl] = useState("");
+  const [title, setTitle] = useState("");
+  const [price, setPrice] = useState(1);
+
+  const mintNft = async (imageUrl, title, price) => {
```

Comment:
Can we move these general functions related to NFTs to a common utils file? In case template user doesn't want to use our frontend (might not be building an nft collection) then they can just keep the utils file containing functions for minting, buying, transferring NFTs etc